docs/blog/_posts/2019-03-01-13th-dotty-milestone-release.md

```diff
@@ -0,0 +1,407 @@
+---
+layout: blog-page
+title: Announcing Dotty 0.13.0-RC1
+author: Aggelos Biboudis
+authorImg: /images/aggelos.jpg
+date: 2019-03-01
+---
+
+Hello hello! This is the second release for 2019, let's call it the _Contextual_
+release and you will understand why we are super excited in a bit! ✨🎊🎉
+
+Without further ado, today we release the version 0.13.0-RC1 of the Dotty compiler.
+This release serves as a technology preview that demonstrates new language features and the
+compiler supporting them.
+
+Dotty is the project name for technologies that are considered for inclusion in Scala 3. Scala has
+pioneered the fusion of object-oriented and functional programming in a typed setting. Scala 3 will
+be a big step towards realising the full potential of these ideas. Its main objectives are to
+
+- become more opinionated by promoting programming idioms we found to work well,
+- simplify where possible,
+- eliminate inconsistencies and surprising behaviours,
+- build on strong foundations to ensure the design hangs well together,
+- consolidate language constructs to improve the language’s consistency, safety, ergonomics, and
+  performance.
+
+You can learn more about Dotty on our [website](https://dotty.epfl.ch).
+
+<!--more-->
+
+This is our 132th scheduled release according to our
+[6-week release schedule](https://dotty.epfl.ch/docs/usage/version-numbers.html).
+
+# What’s new in the 0.13.0-RC1 technology preview?
+
+## All things impl... implied
+
+Scala's implicits are its most distinguished feature. They are _the_ fundamental
+way to abstract over context. They represent a single concept with an extremely
+varied number of use cases, among them: implementing type classes, establishing
+context, dependency injection, expressing capabilities, computing new types and
+proving relationships between them.
+
+However, we identify a few side-effects that implicits gave rise to, as a
+programming style. Firstly, users used implicit conversions between types, in an
+undisciplined matter. This overuse of implicit conversions declattered code for
+sure, but it made it harder for people to reason about.
+
+```scala
+implicit def i1(implicit x: T): C[T] = ... // 1: conditional implicit value
+implicit def i2(x: T): C[T] = ...          // 2: implicit conversion
+```
+
+The `implicit` keyword is used for both implicit conversions and conditional
+implicit values and we identify that their semantic differences must be
+communicated more clearly syntactically. Secondly, implicits pose challenges for
+tooling such as error reporting for failed implicit searches. Furthermore, the
+`implicit` keyword is way too overloaded (implicit vals, defs, objects,
+parameters, or arguments) and users may use it as a mechanism rather than an

```

Comment:
I don't understand the "arguments" here.  There's no use of the `implict` keyword in implicit arguments, isn't there?

Reply:
How was this resolved?  I still see a reference to use of the `implicit` keyword and arguments.

Reply:
I was thinking it is in the same sense as in: 

https://docs.scala-lang.org/sips/byname-implicits.html

Reply:
But maybe you are right, we need to align the terminology. I am removing.

Reply:
Looks like that text is wrong.  Those are implicit parameters.  In fact it refers to them sometimes as "parameters" too.  The spec seems to do a good job using the right terms: https://www.scala-lang.org/files/archive/spec/2.11/07-implicits.html#implicit-parameters

I'd suggest just dropping "arguments": there are only implicit definitions (val/def/object), the providers, and implicit parameters, the consumers.

Reply:
Oh I just saw your push.  :+1: all done, thanks.

Reply:
You were right! Thanks for pointing it out again. I missed it.